extensions/BugModal/web/new_bug.js

```diff
@@ -1,30 +1,78 @@
 var initial = {}
 var comp_desc = {}
+var product_name = '';
+
+var component_load = function(product) {
+    $('#product-throbber').show();
+    $('#component').attr('disabled', true);
+    bugzilla_ajax(
+        {
+            url: 'rest/bug_modal/product_info?product=' + encodeURIComponent(product)
+        },
+        function(data) {
+            $('#product-throbber').hide();
+            $('#component').attr('disabled', false);
+            $('#comp_desc').text('Select a component to read its description.');
+            var selectize = $("#component")[0].selectize;
+            selectize.clear();
+            selectize.clearOptions();
+            selectize.load(function(callback) {
+                callback(data.components)
+            });
+
+            for (var i in data.components)
+                comp_desc[data.components[i]["name"]] = data.components[i]["description"];
+
+            selectize = $("#version")[0].selectize;
+            selectize.clear();
+            selectize.clearOptions();
+            selectize.load(function(callback) {
+                callback(data.versions);
+            });
+        },
+        function() {
+            alert("Network issues. Please refresh the page and try again");
+        }
+    );  
+}
 
 $(document).ready(function() {
+    var current_hash = window.location.hash;
```

Comment:
Wouldn't it be better, if we don't go with LowerCase? It is used by `component_load()` and `$("#product").value()` which interact with backend cgi directly. To get lower case hash in url covered, I can just put: 
```
if (initial.products[product].name.LowerCase() === current_hash.substr(1).LowerCase()) 
```